src/Tribe/Views/V2/Views/Traits/HTML_Cache.php

```diff
@@ -253,13 +253,20 @@ public function get_cache_html_key() {
 	 * @return bool
 	 */
 	protected function should_enable_html_cache( $context ) {
+		global $post;
+
 		$event_date = $context->get( 'event_date' );
 
 		// Respect the month view caching setting.
 		if ( ! tribe_get_option( 'enable_month_view_cache', true ) ) {
 			return false;
 		}
 
+		// Don't cache pages using the `tribe_this_week` shortcode to allow the layout setting to be applied.
+		if ( $post instanceof \WP_Post && has_shortcode( $post->post_content, 'tribe_this_week' ) ) {
+			return false;
+		}
```

Comment:
This code is to support something in the Pro Shortcodes, we need to move it to the Pro codebase and utilize the `tribe_events_views_v2_should_cache_html` filter to allow third party developers to modify it. The hook i mentioned here already exists.